cli/cmd/component-delete.go

```diff
@@ -105,17 +91,48 @@ func runDelete(cmd *cobra.Command, args []string) {
 		contextLogger.Fatal("Suitable kubeconfig file not found. Did you run 'lokoctl cluster apply' ?")
 	}
 
-	if err := deleteComponents(kubeconfig, componentsObjects...); err != nil {
+	if err := deleteComponents(kubeconfig, componentObjects); err != nil {
 		contextLogger.Fatal(err)
 	}
 }
 
-func deleteComponents(kubeconfig []byte, componentObjects ...components.Component) error {
+// componentNames returns list of components to operate on. If explicit list is empty,
+// it returns components defined in the configuration.
+func componentNames(list []string, lokomotiveConfig config.Config) []string {
+	if len(list) != 0 {
+		return list
+	}
```

Comment:
This function does a lot more than just extract names of components. I think it makes it harder for the user of this function to know all the magic that goes behind it.

I would argue that let the caller take care of checking length and all and decide on where to extract names from.

Reply:
Given following use:
```go
componentsToDelete := componentNames(args, *lokoConfig)
```

IMO it is clear, that `componentNames` selects `componentsToDelete` based on `args` and `lokoConfig`.

Perhaps renaming it to `selectComponentNames` would make more sense?

Reply:
I think so `selectComponentNames` should make it more explicit yes.

Reply:
Changed.